### PR TITLE
#481 Improve OCR Quality/Accuracy

### PR DESCRIPTION
**Pull request by breezko**
Fixing https://github.com/becurrie/titandash/issues/481

Explanation:

+ Use Legacy Mode for OCR and use a decent traineddata file for it.
+ Invert Colors (Tess 4.0+ is better at finding white text on black background)

### CODE REVIEW

File: titanbot/titandash/bot/core/bot.py [hunk 1]
```diff
@@ -1603,6 +1603,11 @@ def purchase_new(image, point, color):
                 return False
 
             artifact = self.next_artifact_upgrade
+            
+            if artifact is None:
```
Comment (becurrie) [inline titanbot/titandash/bot/core/bot.py:1607]: Can you remove this block since it's currently already present in #479

Comment (skvie) [reply titanbot/titandash/bot/core/bot.py]: Yeah, will do

File: titanbot/titandash/bot/core/stats.py [hunk 2]
```diff
@@ -360,6 +360,31 @@ def stage_ocr(self, test_image=None):
...
+
+        image = np.array(image)
+        # Resize image.
+        image = cv2.resize(image, None, fx=scale, fy=scale, interpolation=cv2.INTER_CUBIC)
+            
+        # define range of white color in HSV
+        # change it according to your need !
+        lower_white = np.array([0,0,0], dtype=np.uint8)
+        upper_white = np.array([0,0,255], dtype=np.uint8)
+        hsv = cv2.cvtColor(image,cv2.COLOR_BGR2HSV)
```
Comment (skvie) [inline titanbot/titandash/bot/core/stats.py:379]: added hsv filter to custom function

File: titanbot/titandash/bot/core/stats.py [hunk 3]
```diff
@@ -374,11 +399,11 @@ def get_advance_start(self, test_image=None):
             image = self._process_stage(test_image)
         else:
             self.grabber.snapshot(region=region)
-            image = self._process(region=region)
+            image = self._preprocess_stage(scale=5)
```
Comment (skvie) [inline titanbot/titandash/bot/core/stats.py:402]: use the custom function for truly color inversion and better ocr quota

**Comment by becurrie** [comment]
Closing since #509 replicates with some added functionality.

**Pull request closed without merging**